ironfish/src/account/accounts.ts

```diff
@@ -590,18 +590,27 @@ export class Accounts {
     transactionFee: bigint,
     memo: string,
     receiverPublicAddress: string,
+    defaultTransactionExpirationSequenceDelta: number,
+    expirationSequence?: number | null,
   ): Promise<Transaction> {
     const heaviestHead = this.chain.head
     if (heaviestHead === null) {
       throw new ValidationError('You must have a genesis block to create a transaction')
     }
 
+    expirationSequence =
+      expirationSequence ?? heaviestHead.sequence + defaultTransactionExpirationSequenceDelta
+    if (this.chain.verifier.isExpiredSequence(expirationSequence)) {
+      throw new ValidationError('Invalid expiration sequence for transaction')
+    }
+
```

Comment:
I think we'll want to make a change to `rebroadcastTransactions` elsewhere in this file to remove the transaction from the accounts store if it's past the expiration sequence. This should fix the notes permanently appearing as unspendable in the wallet, as well as continuing to rebroadcast transactions that other nodes will reject as invalid.

I'm okay adding this in another PR since this one is fairly large.